pages/_document.js

```diff
@@ -23,40 +12,24 @@ export default class Document extends NextDocument {
         <Head />
         <body>
           <Main />
-          {process.env.NODE_ENV !== "production" ? (
+          <template id="scripts">
             <NextScript />
-          ) : (
-            <script
-              dangerouslySetInnerHTML={{
-                __html: (() => {
-                  const source = renderToStaticMarkup(
-                    NextDocument.renderDocument(Scripts, this.props)
-                  )
-                    .split("</script>")
-                    .filter(Boolean)
-                    .map((s) =>
-                      JSON.stringify(encodeURIComponent(s + "</script>"))
-                    );
-                  const script = `
-                    if (!window.location.search.includes('no_script')) {
-                        [${source}].map(decodeURIComponent).forEach(s => {
-                            const parser = new DOMParser();
-                            const doc = parser.parseFromString(s, "text/html");
-                            const c = doc.head.children[0];
-                            const e = document.createElement('script');
-                            for (const n of c.getAttributeNames()) {
-                                e.setAttribute(n, c.getAttribute(n));
-                                e.innerHTML = c.innerHTML;
-                            }
-                            document.body.appendChild(e);
-                        });
-                    }
-                  `;
-                  return script.trim().replace(/\n/gm, "").replace(/\s+/g, " ");
-                })(),
-              }}
-            />
-          )}
+          </template>
+          <script
+            dangerouslySetInnerHTML={{
+              __html: (() =>
+                `
+                if (!window.location.search.includes('no_script')) {
+                  const template = document
+                    .querySelector("#scripts")
+                    .content.cloneNode(true);
+                  Array.from(template.querySelectorAll("script")).forEach((script) => {
+                    document.body.appendChild(script);
```

Comment:
Rendering the scripts is pretty straightforward:
- Get the template.
- Clone its content.
- Query the script tags within it.
- Iterate over them, and append them all to the body.